Google authentication landing page

nikhen-s opened the pull request:
## Objective
Implement Google Authentication for users to be able to view and modify required information to use the app

## Key Changes
- [x] Aligned the Login Page UI to look like the Figma Mockup
- [x] Button on the landing page successfully calls the Google Authentication API
- [x] Upon successful authentication, the app navigates to home page
- [x] Upon failing authentication, the app navigates back to login page (and shows an error message in a AlertDialog Box)

Related to Issue https://github.com/swent-sp-2024-event-radar/event-radar-app/issues/13

jgeaso1266 commented on app/google-services.json:
```diff
@@ -12,15 +12,33 @@
         }
       },
-      "oauth_client": [],
+      "oauth_client": [
```
It looks like these changes are still in this PR, can you move this to your other PR?

nikhen-s replied:
Done, Files Changed now doesn't show this :)

jgeaso1266 commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -33,17 +46,35 @@ import com.github.se.eventradar.ui.navigation.NavigationActions
+@SuppressLint("UnrememeredMutableState", "UnrememberedMutableState")
+@Composable
+fun LoginScreen(navigationActions: NavigationActions) {
+  val openErrorDialog = remember { mutableStateOf(false) }
```
The standard way to do this is using the `by` keyword (then you won't need that `@SuppressLint` annotation.

nikhen-s replied:
DM'ed, and solution is ok it turns out 👍

jgeaso1266 commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -55,57 +86,94 @@ fun LoginScreen(navigationActions: NavigationActions) {
+                  ))
+        }
+
+    Spacer(modifier = Modifier.height(240.dp))
```
nit: it might be better to use constraints instead of just dp values to adapt to different screen sizes.

nikhen-s replied:
Changed to constraints layout in latest commit.

jgeaso1266 commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -55,57 +86,94 @@ fun LoginScreen(navigationActions: NavigationActions) {
+                      lineHeight = 17.sp,
+                      fontFamily = FontFamily(Font(R.font.roboto)),
+                      fontWeight = FontWeight(700),
+                      color = Color(0xFF000000),
```
nit: the color should be using the material design options like `primary` and `primaryContainer` as opposed to hardcoding.

nikhen-s replied:
Done as well, just waiting for Vicky's PR to be merged so that I can use primaryContainer colour.

jgeaso1266 commented on app/src/main/res/layout/activity_firebase_uiactivity.xml:
```diff
@@ -3,7 +3,6 @@
-    tools:context=".SignInActivity">
-
+    android:layout_height="match_parent">
+<!--    tools:context=".SignInActivity"> -->
```
nit: I don't believe this needs to be commented out (even though it was red). This is generated by firebase I believe.

nikhen-s replied:
Done :)

jgeaso1266 commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,6 +1,7 @@
 <resources>
     <string name="app_name">My Application</string>
-    <string name="web_client_id">957415054330-j5fa1cl19qlthoshv7l2cccqv5v4icj4.apps.googleusercontent.com</string>
+    <string name="web_client_id">1043895872028–8m8f4kpcmqe60ofkef734vbgqph8g5ut.apps.googleusercontent.com</string>
```
This should move to the other PR as well

nikhen-s replied:
👍

jgeaso1266 commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -55,57 +86,94 @@ fun LoginScreen(navigationActions: NavigationActions) {
+    Row(
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically) {
+          Image(
```
You should modify the color to work with both light and dark mode (using the material design colors).

nikhen-s replied:
👍

christinehahaa commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
Nice!

christinehahaa commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -33,17 +46,35 @@ import com.github.se.eventradar.ui.navigation.NavigationActions
+            }
           })
-
+  ErrorDialogBox(openErrorDialog)
```
Nice!

Viccha18 replied:
Good additon to handle the failure case

christinehahaa commented on app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:
```diff
@@ -33,17 +46,35 @@ import com.github.se.eventradar.ui.navigation.NavigationActions
+        title = { Text("Sign in Failed") },
+        onDismissRequest = { openErrorDialog.value = false },
+        confirmButton = {
+          TextButton(onClick = { openErrorDialog.value = false }) { Text("Confirm") }
```
nitpick: I'm thinking "Ok" instead of "Confirm" makes more sense? We didn't have a Figma mockup for this, but it's a small thing so no need for one.

nikhen-s replied:
👍

Viccha18 reviewed (approved):
Good job ! Nothing to add from Jalen and Christine. Just a reminder for later : the new route to homescreen will be `Route.EVENTS` after we merged the new navigation bar PR

nikhen-s commented:
Newest Commits: All the changes are made, 
Vicky's highlight on the new route to homescreen will be Route.EVENTS, I've added as a comment next to Route.Overview since waiting for Navigation Bar PR to merge to main + Japen's highlight on using Material Theme Colours, I have changed everything with exception for the primaryContainer for button since waiting on Vicky's Colours PR to merge to main.
